src/main/haskell/matching-logic/test/Kore/MatchingLogic/ProofSystem/OnePlusOne.hs

```diff
@@ -0,0 +1,446 @@
+import qualified Data.Map.Strict                        as Map
+import qualified Data.Map.Strict                        as Map
+import qualified Data.Set                               as Set
+import           Data.Text                              (Text)
+import           Data.Text.Prettyprint.Doc              hiding (parens,space)
+import qualified Data.Tree                              as Tree
+import           Data.Word
+
+import           GHC.Generics
+import           Data.Data
+
+import           Control.Lens
+import           Data.Functor.Foldable                  (Fix (Fix))
+import           Data.Void(Void)
+
+import           Text.Megaparsec(Parsec,tokens,parse,parseErrorPretty,eof)
+import           Text.Megaparsec.Byte
+import qualified Text.Megaparsec.Byte.Lexer as Lexer
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+import qualified Kore.MatchingLogic.ProofSystem.OnePlusOne.ForallAST as AST
+import           Kore.MatchingLogic.HilbertProof        as HilbertProof (Proof (..),
+                                                                         ProofSystem,
+                                                                         add,
+                                                                         derive,
+                                                                         emptyProof)
+import           Kore.MatchingLogic.ProofSystem.OnePlusOne.ProofSystem ( MLRule (..)
+                                                        , MLRuleSig
+                                                        , transformRule
+                                                        , SubstitutedVariable (..)
+                                                        , SubstitutingVariable (..))
+import           Kore.MatchingLogic.Signature.Simple
+
+-- START code extracted by Coq
+data Nat =
+   O
+ | S Nat
+  deriving (Read,Show,Eq,Generic,Data)
+
+data Formula =
+   Plus Formula Formula
+ | Succ Formula
+ | Zero
+ | Not Formula
+ | And Formula Formula
+ | Var Nat
+ | All Nat Formula
+ | Defined Formula
+  deriving (Read,Show,Eq,Generic,Data)
+
+data Ctx =
+   SuccC
+ | PlusLC Formula
+ | PlusRC Formula
+ | DefinedC
+  deriving (Read,Show,Eq,Generic,Data)
+
+data Ctx_path =
+   Path_nil
+ | Path_cons Ctx Ctx_path
+  deriving (Read,Show,Eq,Generic,Data)
+
+data Simple_rule =
+   Rule_mp Simple_proof Simple_proof
+ | Rule_frame Ctx Simple_proof
+ | Rule_gen Nat Simple_proof
+ | Rule_use_hyp Nat
+ | Ax_propositional1 Formula Formula
+ | Ax_propositional2 Formula Formula Formula
+ | Ax_propositional3 Formula Formula
+ | Ax_varSubst Nat Nat Formula Formula
+ | Ax_allImp Nat Formula Formula
+ | Ax_out_ex Ctx Nat Formula
+ | Ax_out_or Ctx Formula Formula
+ | Ax_existence Nat
+ | Ax_singvar Ctx_path Ctx_path Nat Formula
+ | Ax_definedness
+ | Ax_proj1 Formula Formula
+ | Ax_proj2 Formula Formula
+ | Ax_and_intro Formula Formula
+ | Ax_or_elim Formula Formula Formula
+ | Ax_or_intro1 Formula Formula
```

Comment:
A text description of these rules would be nice. I mean, I know some of them, e.g. Modus Ponens, but I think that this is the first time I see, say, Ax_or_intro1. That text description can be in the ProofSystem.hs file, but then it would be nice to have a pointer there. Also, why is there a naming difference between these and the ProofSystem stuff (besides the `Ax` prefix)?